File: output/mistralai_mistral_small_2603/few/cpp/Torre.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;

    vector<vector<int>> board(n, vector<int>(n));
    vector<long long> row_sum(n, 0), col_sum(n, 0);

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cin >> board[i][j];
            row_sum[i] += board[i][j];
            col_sum[j] += board[i][j];
        }
    }

    long long max_weight = 0;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            long long weight = row_sum[i] + col_sum[j] - board[i][j];
            if (weight > max_weight) {
                max_weight = weight;
            }
        }
    }

    cout << max_weight << endl;
    return 0;
}

File: output/cbsoft_sbes_2026/anthropic_claude_sonnet_4_6/zero/python_noimg/Torre.py
import sys
input = sys.stdin.readline

def main():
    n = int(input())
    board = []
    for _ in range(n):
        row = list(map(int, input().split()))
        board.append(row)
    
    row_sum = [sum(board[i]) for i in range(n)]
    col_sum = [sum(board[i][j] for i in range(n)) for j in range(n)]
    
    max_weight = 0
    for i in range(n):
        for j in range(n):
            weight = row_sum[i] + col_sum[j] - board[i][j]
            if weight > max_weight:
                max_weight = weight
    
    print(max_weight)

main()